mono/utils/mono-threads.h

```diff
@@ -710,4 +719,39 @@ typedef void (*background_job_cb)(void);
 void mono_threads_schedule_background_job (background_job_cb cb);
 #endif
 
+typedef struct {
+	void (*init) (gpointer *state_ptr);
+	void (*command) (gpointer state_ptr, gpointer message_ptr);
+	void (*cleanup) (gpointer state_ptr);
+} MonoUtilityThreadCallbacks;
+
+typedef struct {
+	MonoNativeThreadId thread_id;
+
+	MonoLockFreeQueue work_queue;
+	MonoSemType work_queue_sem;
+	gboolean run_thread;
+
+	MonoLockFreeAllocator message_allocator;
+	MonoLockFreeAllocSizeClass message_size_class;
+
+	unsigned int message_block_size;
+	unsigned int payload_size;
+
+	gpointer state_ptr;
+	MonoUtilityThreadCallbacks callbacks;
+} MonoUtilityThread;
+
+MonoUtilityThread *
+mono_utility_thread_launch (size_t payload_size, MonoUtilityThreadCallbacks *callbacks);
```

Comment:
These would also need to be marked `MONO_PROFILER_API` for the profiler to use them.